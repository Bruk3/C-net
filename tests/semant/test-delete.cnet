struct person {
    string name;
};

int main(){
    struct person john;
    struct person jane;
    int[] myInts;
    myInts = new int[4]{1,2,3,4};
    string[] myStrings = new string[2]{"hello", "world"};
    struct person[] myPeople = new struct person[2]{john, jane};
    delete myPeople[0];
    delete myPeople;
    delete john;
    delete myInts;
}