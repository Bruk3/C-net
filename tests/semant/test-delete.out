struct person {
	string name;
};

int main()
{
struct person john;
struct person jane;
int[] myInts;
(int[] : myInts=(int[] : new int[(int : 4)] = {(int : 1), (int : 2), (int : 3), (int : 4)}));
string[] myStrings = (string[] : new string[(int : 2)] = {(string : "hello"), (string : "world")});
struct person[] myPeople = (struct person[] : new struct person[(int : 2)] = {(struct person : john), (struct person : jane)});
delete (struct person : myPeople[0]);
delete (struct person[] : myPeople);
delete (struct person : john);
delete (int[] : myInts);
}

